### DISCUSSION TITLE
Continuous build 0686025 Slider SV problem

### DISCUSSION THREAD

**Discussion by Terracotta-6** [artisan-roaster-scope/artisan#1527] [General]
![b66a822382147fe8bb77af2e4c2d4cf](https://github.com/artisan-roaster-scope/artisan/assets/149906183/311fed99-ec07-4b40-af73-bb10853c2f50)
![afdd303635633323eac10f20dee7c16](https://github.com/artisan-roaster-scope/artisan/assets/149906183/23db596e-47ad-4000-a3c4-6d8d80670f46)
![aa529c8758c13ac0cc3cd2d18000f0c](https://github.com/artisan-roaster-scope/artisan/assets/149906183/8dfe2f99-b7d2-4080-9082-af0c66aa92d4)

**Comment by MAKOMO** [comment]
Could you please attach your initial settings such that I have a chance to reproduce this? Thanks!

**Reply by Terracotta-6** [reply]
Modbus Error: writeSingleRegister() argument out of range@line 590
Roaster Scope

**Reply by MAKOMO** [reply]
Nothing attached. Sorry. Could you proper upload those settings via drag-and-drop after renaming to end in *.txt? Thanks!

**Comment by Terracotta-6** [comment]
The slider Max value is reset frequently "110C"after starting artisan.
![b6b5a5e91dbad84bea7037011542517](https://github.com/artisan-roaster-scope/artisan/assets/149906183/a1e389fc-5d1c-4aff-9785-c199d7e7e195)

**Comment by Terracotta-6** [comment]
Simply build the hardware temperature controller and test with mudbus RTU protocol，The test results are the same problem,Not related to the TC4 protocol.

**Reply by MAKOMO** [reply]
There is no need to build hardware. A software MODBUS simulator will do. Indeed the mixed TC4 setup seems not to play a role here as the MODBUS PID specification takes precedence over the Arduino/TC4 external PID (as can see from the title in the PID Dialog).

I tried everything, but I cannot reproduce what I see in your screenshots. The only thing that seems to be wrong is the -18 while the limits are set to 0-220. I couldn't reproduce the change of the max SV slider limit on restart or by any other means.

Could you provide a step by step description incl. a description what you think is wrong. Thanks a lot!

**Reply by Terracotta-6** [reply]
[Test1-artisan-settings.aset.txt](https://github.com/artisan-roaster-scope/artisan/files/14877799/Test1-artisan-settings.aset.txt)

**Reply by Terracotta-6** [reply]
https://github.com/artisan-roaster-scope/artisan/assets/149906183/00d4c013-1831-45ea-ad1d-e398b4a26ea7

**Reply by Terracotta-6** [reply]
The maximum SV slider was occasionally changed under the TC4 protocol, and has not been successfully repeated today.

**Comment by MAKOMO** [comment]
I think I fixed this one. It seems that switching temperature units between C and F could lead to slider limits beyond the programmed min/max of 0-999. Commit 3487ab01e6661fb8a8e1beaec5b290d3cb78151f enforces those limits all over the app.

**Reply by Terracotta-6** [reply]
Perfect! Thank you.
